domain/test/usecases/download/download_file_interactor_test.dart

```diff
@@ -0,0 +1,91 @@
+//  3 and <http://www.linshare.org/licenses/LinShare-License_AfferoGPL-v3.pdf> for
+//  the Additional Terms applicable to LinShare software.
+//
+
+import 'package:dartz/dartz.dart';
+import 'package:domain/domain.dart';
+import 'package:mockito/mockito.dart';
+import 'package:test/test.dart';
+
+import '../../fixture/test_fixture.dart';
+import '../../mock/repository/authentication/mock_credential_repository.dart';
+import '../../mock/repository/authentication/mock_document_repository.dart';
+import '../../mock/repository/authentication/mock_token_repository.dart';
+
+void main() {
+  group('download_file_interactor_test', () {
+    MockDocumentRepository documentRepository;
+    MockTokenRepository tokenRepository;
+    MockCredentialRepository credentialRepository;
+    DownloadFileInteractor downloadFileInteractor;
+    DocumentId documentId;
+
+    setUp(() {
+      documentRepository = MockDocumentRepository();
+      tokenRepository = MockTokenRepository();
+      credentialRepository = MockCredentialRepository();
+      documentId = DocumentId('383-384-C');
+      downloadFileInteractor = DownloadFileInteractor(documentRepository, tokenRepository, credentialRepository);
+    });
+
+    test('downloadFileInteractor should return success with correct document ID', () async {
+      when(tokenRepository.getToken()).thenAnswer((_) async => permanentToken);
+      when(credentialRepository.getBaseUrl()).thenAnswer((_) async => linShareBaseUrl);
+      when(documentRepository.downloadDocument(documentId, permanentToken, linShareBaseUrl)).thenAnswer((_) async => DownloadTaskId('task_id_1'));
+
+      final result = await downloadFileInteractor.execute(documentId);
+
+      expect(result, Right<Failure, Success>(DownloadFileViewState(DownloadTaskId('task_id_1'))));
+    });
+
+    test('downloadFileInteractor should failure with wrong baseUrl', () async {
+      when(tokenRepository.getToken()).thenAnswer((_) async => permanentToken);
+      when(credentialRepository.getBaseUrl()).thenAnswer((_) async => wrongUrl);
+      final exception = Exception();
+      when(documentRepository.downloadDocument(documentId, permanentToken, wrongUrl)).thenThrow(exception);
+
+      final result = await downloadFileInteractor.execute(documentId);
+
+      expect(result, Left<Failure, Success>(DownloadFileFailure(exception)));
+    });
+
+    test('downloadFileInteractor should failure with wrong token', () async {
+      when(tokenRepository.getToken()).thenAnswer((_) async => wrongToken);
+      when(credentialRepository.getBaseUrl()).thenAnswer((_) async => linShareBaseUrl);
+      final exception = Exception();
+      when(documentRepository.downloadDocument(documentId, wrongToken, linShareBaseUrl)).thenThrow(exception);
+
+      final result = await downloadFileInteractor.execute(documentId);
+
+      expect(result, Left<Failure, Success>(DownloadFileFailure(exception)));
+    });
+  });
+}
```

Comment:
How about `getToken` failed, or `getBaseUrl` failed?

Reply:
Maybe we need to update getBaseUrl to throw an exception if it's null because right now it will never fails

Reply:
Srr, I can not get your point. But why you are sure `getBaseUrl` will not fail? How about when share preference is deleted, and you can not get the url -> parse will be throw?

Reply:
Yes you are right, I covered these cases now

Reply:
I want to verify `getBaseUrl` and `getToken` is invoked 1 time. Please add that verify to all the tests.

Reply:
Check interactors again please I used `verify`